core/embed/extmod/modtrezorui/display.c

```diff
@@ -426,27 +431,35 @@ bool display_toif_info(const uint8_t *data, uint32_t len, uint16_t *out_w,
   return true;
 }
 
-#if defined TREZOR_MODEL_T
+#if defined TREZOR_MODEL_T || defined TREZOR_MODEL_R
 #include "loader.h"
+#ifdef TREZOR_MODEL_R
+#define LOADER_SIZE_ADJ 3
+#else
+#define LOADER_SIZE_ADJ 1
+#endif
 #endif
 
 void display_loader(uint16_t progress, bool indeterminate, int yoffset,
                     uint16_t fgcolor, uint16_t bgcolor, const uint8_t *icon,
                     uint32_t iconlen, uint16_t iconfgcolor) {
-#if defined TREZOR_MODEL_T
+#if defined TREZOR_MODEL_T || defined TREZOR_MODEL_R
```

Comment:
I'd rather keep the `display_loader` only defined for `TREZOR_MODEL_T` and keep it undefined for other models for now. Suggested changes are okayish, but the code is very hard to understand even before the change.